src/audio/copier/copier.c

```diff
@@ -960,8 +932,8 @@ static int copier_reset(struct comp_dev *dev)
 			if (cd->hd->chan)
```

Comment:
ok, you do have a comment above. So it isn't a blocker, but I'd still rather put that outside of the loop

Reply:
at this patch stage, host dev is still not removed, unregister callback still need host dev involved, so move outside   
may broken the code logic, at the final patch, we can move outside, if you check, code already move outside loop

Reply:
Also, this is an complex patch, change too frequently may cause potential bugs, this is my worry, CI also report all kinds of errors.